qcodes/instrument/channel.py

```diff
@@ -543,6 +578,7 @@ def print_readable_snapshot(self, update: bool = False,
                 channel.print_readable_snapshot(update=update,
                                                 max_chars=max_chars)
 
+collections.abc.Sequence.register(ChannelList)
```

Comment:
so this makes ChannelList a Sequnce subclass without inheriting, right? and why do we not want or cannot inherit? because then "isinstance and issubclass" won't work as expected?

Reply:
Mostly this seems to be recommended for existing classes to avoid introducing any unwanted api changes but in this case it would probably be ok

Reply:
@astafan8 pushed a version that implements this as a subclass. This revealed that index was not implemented fully correctly so I guess its an improvement